### ISSUE TITLE
torch.symeig() returns different eigenvectors than numpy.linalg.eigh()

### ISSUE THREAD

**Issue by themightyoarfish** [pytorch/pytorch#16072]
I am not sure if this is a bug (my algebra fu is kinda rusty - are the eigenvectors for given eigenvalues unique? I think not), but I have found at least one case where the eigenvectors returned by `torch.symeig()` are identical to those of `np.linalg.eigh()` only up to sign (some signs differ).

Is this expected behaviour or does PyTorch intend to follow numpy?

I can upload the test case if desired.

**Comment by vishwakftw** [comment]
The eigenvector for a given eigenvalue need not be unique; if v is a eigenvector for a given eigenvalue u, then x.v is also a eigenvector for the same eigenvalue for non-zero scalar x. This is why these eigenvectors are normalized.

In your case, the sign of the entries in the eigenvector could have been toggled (+ to - and vice-versa), but the absolute value of the entries should be the same. The toggling corresponds to x = -1 in the paragraph above.

**Comment by themightyoarfish** [comment]
Of course, scalar multiplication will always result in an eigenvector. I should have been more clear I think; what I'm seeing is that not every eigenvector computed by torch is a scalar multiple of one computed by numpy. Instead, sometimes signs of some entries of an eigenvector are flipped w.r.t. the numpy version, but not the entire vector.
